judgments/forms/search_forms.py

```diff
@@ -21,20 +24,46 @@ def _get_choices_by_group(courts):
         "key2": "value2"
     }
     """
-    options: dict = {}
+    options: dict[Union[Optional[str], CourtParam], Union[str, dict[CourtParam, str]]] = {}
     for group in courts:
         if group.display_heading:
-            option = {group.name: {court.canonical_param: court.grouped_name for court in group.courts}}
+            option_a: dict[Union[Optional[str], CourtParam], Union[str, dict[CourtParam, str]]] = {
+                group.name: {
+                    court.canonical_param: court.grouped_name for court in group.courts if court.canonical_param
+                }
+            }
+            options = options | option_a
         else:
-            option = {court.canonical_param: court.grouped_name for court in group.courts}
-        options = options | option
+            option_b: dict[Union[Optional[str], CourtParam], Union[str, dict[CourtParam, str]]] = {
+                court.canonical_param: court.grouped_name for court in group.courts if court.canonical_param
+            }
+            options = options | option_b
     return options
 
 
 COURT_CHOICES = _get_choices_by_group(all_courts.get_grouped_selectable_courts())
 TRIBUNAL_CHOICES = _get_choices_by_group(all_courts.get_grouped_selectable_tribunals())
 
 
+class CourtTribunalField(forms.MultipleChoiceField):
```

Comment:
I'm not 100% sold on this name, it kinda implies it's the `Tribunal` belonging to a `Court`, which makes no sense.

It's really some kind of horrible abstract `JudicialBodyField`... but I don't like that either.

Reply:
Changed to CourtOrTribunalField